addons/pos_restaurant/static/src/app/floor_screen/floor_screen.xml

```diff
@@ -26,8 +26,11 @@
                     <i class="fa fa-pencil-square-o" role="img" aria-label="Rename" title="Rename" />Rename</button>
                 <button t-attf-class="{{editButtonClass}}" t-on-click.stop="duplicateTableOrFloor">
                     <i class="fa fa-clone" role="img" aria-label="Copy" title="Copy" />Copy</button>
-                <button t-attf-class="{{editButtonClass}}" t-if="activeTables.some(t=>t.parent_id)" t-att-disabled="!selectedTables.some(t=>t.parent_id)" t-on-click="unlinkTables">
+                <button t-attf-class="{{editButtonClass}}" t-if="selectedTables.length and selectedTables.every(t=>t.parent_id)" t-on-click="unlinkTables">
                     <i class="fa fa-unlink" role="img" aria-label="Unlink" title="Unlink" />Unlink</button>
+                <!--                                                 Conditions for tables to be linked:           ↓↓↓   all the selected tables must have the same parent or no parent      ↓↓↓   ↓↓↓among the tables there can only be one that has children↓↓↓              -->
+                <button t-attf-class="{{editButtonClass}}" t-else="" t-att-disabled="selectedTables.length lt 2 or unique(selectedTables.filter(t=>t.parent_id).map(t=>t.parent_id)).length gt 1 or selectedTables.filter(t=>getChildren(t).length).length gt 1" t-on-click="linkTables">
```

Comment:
you should extract this commend and the value for t-att-disabled as a method in the component.